src/inc/til/rectangle.h

```diff
@@ -405,6 +412,254 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
+            // The sign determines which edge of the rectangle moves.
+            // We use the magnitude as how far to move.
+            if (width > 0)
+            {
+                // Subtracting the positive makes the rectangle "shrink"
+                // because right pulls inward (to the left).
+                //
+                // Example with subtracting width 3...
+                // |-- x = origin
+                // V
+                // x---------|    x------|
+                // |         |    |      |
+                // |         |    |      |
+                // |---------|    |------|
+                // BEFORE         AFTER
+                THROW_HR_IF(E_ABORT, !base::CheckSub(r, width).AssignIfValid(&r));
+            }
+            else
+            {
+                // Subtracting the negative makes the rectangle "shrink"
+                // because left pulls inward (to the right).
+                //
+                // Example with subtracting width -3...
+                // |-- x = origin
+                // V
+                // x---------|    x  |------|
+                // |         |       |      |
+                // |         |       |      |
+                // |---------|       |------|
+                // BEFORE         AFTER
+                THROW_HR_IF(E_ABORT, !base::CheckSub(l, width).AssignIfValid(&l));
+            }
+
+            if (height > 0)
+            {
+                // Subtracting the positive makes the rectangle "shrink"
+                // because bottom pulls inward (to the up).
+                //
+                // Example with subtracting height 2...
+                // |-- x = origin
+                // V
+                // x---------|    x---------|
+                // |         |    |---------|
+                // |         |
+                // |---------|
+                // BEFORE         AFTER
+                THROW_HR_IF(E_ABORT, !base::CheckSub(b, height).AssignIfValid(&b));
+            }
+            else
+            {
+                // Subtracting the positive makes the rectangle "shrink"
+                // because top pulls inward (to the down).
+                //
+                // Example with subtracting height -2...
+                // |-- x = origin
+                // V
+                // x---------|    x
+                // |         |
+                // |         |    |---------|
+                // |---------|    |---------|
+                // BEFORE         AFTER
+                THROW_HR_IF(E_ABORT, !base::CheckSub(t, height).AssignIfValid(&t));
+            }
+
+            return rectangle{ til::point{ l, t }, til::point{ r, b } };
+        }
+
+        rectangle& operator-=(const size& size)
+        {
+            *this = *this - size;
+            return *this;
+        }
+
+        // MUL will scale the entire rectangle by the size L/R * WIDTH and T/B * HEIGHT.
+        rectangle operator*(const size& size) const

```

Comment:
I'd think that this would leave the origin unmodified, and just increase the w/h by size

Reply:
yeah i am confused here too

Reply:
OK, I'm deleting this one for now. I don't actually have a use for it yet. It was along for the ride when I pulled in stuff from my prototyping branch into this staging one.

Reply:
But what you're asking for (w/h by size) is what add does.